src/storage/v2/replication/replication_client.cpp

```diff
@@ -537,6 +537,33 @@ std::vector<Storage::ReplicationClient::RecoveryStep> Storage::ReplicationClient
   return recovery_steps;
 }
 
+Storage::TimestampInfo Storage::ReplicationClient::GetTimestampInfo() {
+  Storage::TimestampInfo info;
+  info.current_timestamp = 0;
+  info.current_number_of_timestamp_behind_master = 0;
+
+  try {
+    auto stream{rpc_client_->Stream<replication::TimestampRpc>()};
+    const auto response = stream.AwaitResponse();
+    const auto is_success = response.success;
+    if (!is_success) {
+      replica_state_.store(replication::ReplicaState::INVALID);
+      HandleRpcFailure();
+    }
+    auto main_time_stamp = storage_->last_commit_timestamp_.load();
+    info.current_timestamp = response.current_commit_timestamp;
```

Comment:
```suggestion
    info.current_timestamp = main_time_stamp;
```
```suggestion
    info.current_timestamp = response.current_commit_timestamp;
```

Reply:
Original idea was to have info.current_timestamp containing the current timestamp of the replica (that returned the response). I think I disagree with the suggestion here.
(Note: attribute will be renamed for clarity as `current_timestamp` is definitely confusing)